### PR TITLE
updated cw and wasmi deps, cleanup

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
I am on these fork right now, after pallet-cw updated, will remove existing forks in CF, and replace by these.
- for https://github.com/ComposableFi/research/issues/147
- https://github.com/CosmWasm/serde-json-wasm/pull/56
- https://github.com/CosmWasm/cosmwasm/pull/1716
- https://github.com/CosmWasm/cw-storage-plus/pull/26
- updated wasmi
- formatting
- moved all noise(license, developers) to separate files

### CODE REVIEW

File: vm-wasmi/Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
 name = "cosmwasm-vm-wasmi"
-version = "0.1.0"
+version = "0.2.0"
 edition = "2021"
-authors = [
-  "Hussein Ait Lahcen hussein.aitlahcen@gmail.com",
-  "Abdullah Eryuzlu abdullaheryuzlu@gmail.com",
-  "Composable Developers",
-]
-homepage = "https://composable.finance"
```
Comment (hussein-aitlahcen) [inline vm-wasmi/Cargo.toml:10]: interesting

Comment (dzmitry-lahoda) [reply vm-wasmi/Cargo.toml:10]: feel free to join back:) I just dislike noise in headers. I will not add myself sure. Feel free to PR to add these back to the end of toml file

Comment (dzmitry-lahoda) [reply vm-wasmi/Cargo.toml:10]: or add to workspace toml, and there is even tooling to add that during release

Comment (dzmitry-lahoda) [reply vm-wasmi/Cargo.toml:10]: without noise in each file

Comment (dzmitry-lahoda) [reply vm-wasmi/Cargo.toml:10]: no i think useful to point to homepage at all - if somebody operates on source code of vm level - he either wants code, or spec. sure not marketing materials.

Comment (dzmitry-lahoda) [reply vm-wasmi/Cargo.toml:10]: added to readme

Comment (dzmitry-lahoda) [reply vm-wasmi/Cargo.toml:10]: see also SPEC.md file

```
Composable Finance
Hussein Ait Lahcen, Abdullah Eryuzlu, Karel L. Kubat, Antonio M. Larriba
2022-11-04
```

Comment (dzmitry-lahoda) [reply vm-wasmi/Cargo.toml:10]: so who cares if it obsolete

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,4 @@ flamegraph.svg
 *.log
 *.data
 *.data.old
+.vscode/setting.json
```
Comment (mina86) [inline .gitignore]:
```suggestion
.vscode/
```

File: SPEC.md [hunk 3]
```diff
@@ -20,68 +20,71 @@ It describes the semantics of the VM in a slightly more structured way than the
...
-    - [6.2. Host Functions](#62-host-functions)
-        - [6.2.1. IBC Transfer](#621-ibc-transfer)
-        - [6.2.2. IBC Packet Send](#622-ibc-packet-send)
-        - [6.2.3. IBC Channel Close](#623-ibc-channel-close)
+  - [6.1. Enabling IBC](#61-enabling-ibc)
+  - [6.2. Host Functions](#62-host-functions)
+    - [6.2.1. IBC Transfer](#621-ibc-transfer)
+    - [6.2.2. IBC Packet Send](#622-ibc-packet-send)
+    - [6.2.3. IBC Channel Close](#623-ibc-channel-close)
 - [7. Contributors](#7-contributors)
```
Comment (mina86) [inline SPEC.md]: IMO keeping table of contents in Markdown files is counterproductive.  I would just get rid of it.

File: flake.nix [hunk 4]
```diff
@@ -28,6 +28,21 @@
 
         crane-nightly = crane-lib.overrideToolchain rust-nightly;
 
+        check-crate = pkgs.writeShellApplication rec {
+          name = "check-crate";
+          runtimeInputs = [ rust-nightly ];
+          text = ''
+            EXTRA_FEATURES=""
+            if [[ -n "''${2-}" ]]; then
```
Comment (mina86) [inline flake.nix:36]: What’s with the `''`? Shouldn’t this be:

```suggestion
            if [[ -n "${2-}" ]]; then
```

File: vm-wasmi/src/semantic.rs [hunk 5]
```diff
@@ -881,14 +883,21 @@ impl Rules for ConstantCostRules {
...
         wasm_instrument::parity_wasm::elements::Module::from_bytes(code).expect("impossible");
-    let instrumented_module =
-        wasm_instrument::gas_metering::inject(module, &ConstantCostRules, "env")
-            .expect("impossible");
+    let instrumented_module = wasm_instrument::gas_metering::inject(
+        module,
+        host_function::Injector::new(Version1x::ENV_MODULE, Version1x::ENV_GAS),
+        &ConstantCostRules,
+    )
+    .expect("impossible");
```
Comment (mina86) [inline vm-wasmi/src/semantic.rs:900]:
```suggestion
    .unwrap();
```
No point in using `expect` if the message doesn’t provide any information. (In fact, I’m of the opinion that `expect` is barely ever useful).

**Review by mina86 — APPROVED** [review]
LGTM

File: flake.nix (hunk 4 above)
Comment (dzmitry-lahoda) [reply flake.nix:36]: ' ' is escape character to avoid conflict with nix host.

File: SPEC.md (hunk 3 above)
Comment (dzmitry-lahoda) [reply SPEC.md]: fully aggree. will remove.

File: vm-wasmi/src/semantic.rs (hunk 5 above)
Comment (dzmitry-lahoda) [reply vm-wasmi/src/semantic.rs:900]: it was there, will leave as is. first, it provides. author was thinking that is impossible.
so i can rely in it. while with unwrap author was not thinkin at all. second, lines can be broken or no acces, but panic word will be here. easy to find in code.

Comment (mina86) [reply vm-wasmi/src/semantic.rs:900]: If all expect calls use `impossible` message that they don’t make it any easier to find the code.  I also disagree that using unwrap means that author thinks the unwrap never happens.

**Pull request merged**